Cython/Utility/CppConvert.pyx

```diff
@@ -87,14 +97,24 @@ cdef extern from *:
             bint operator!=(const_iterator)
         const_iterator begin()
         const_iterator end()
+        size_t size()
+
+cdef extern from "Python.h":
+    void Py_INCREF(object)
+    list PyList_New(Py_ssize_t size)
+    void PyList_SET_ITEM(object list, Py_ssize_t i, object o)
 
 @cname("{{cname}}")
 cdef object {{cname}}(const cpp_list[X]& v):
-    o = []
+    o = PyList_New(<Py_ssize_t>v.size())
+    cdef Py_ssize_t i = 0
     cdef cpp_list[X].const_iterator iter = v.begin()
     while iter != v.end():
-        o.append(cython.operator.dereference(iter))
+        item = cython.operator.dereference(iter)
+        Py_INCREF(item)
+        PyList_SET_ITEM(o, i, item)
```

Comment:
I think we should add a size check here to make sure that the vector did not grow during iteration.